### PR TITLE
feat(cmake): port googletest into the project

### PR DESCRIPTION
**Pull request by haruhi55**
resolve https://github.com/TiledTensor/TiledCUDA/issues/18 https://github.com/TiledTensor/TiledCUDA/issues/20 https://github.com/TiledTensor/TiledCUDA/issues/17

I am trying to improve the organization of CMakeLists, but I have run into an issue that I haven't been able to resolve yet.

Specifically, when compiling with nvcc, I need to explicitly set `std=c++17` as a compiling option. If I don't do this, I get errors when compiling with torchlibs that require the c++ standard to be set to 17.

### CODE REVIEW

File: cmake/dependencies.cmake
Comment (KuangjuX) [inline cmake/dependencies.cmake:1]: I have a suggestion to place the `cmake` directory under the `scripts` directory. What do you think about it?

**Review by KuangjuX — CHANGES_REQUESTED** [review]

File: cmake/dependencies.cmake
Comment (lcy-seso) [reply cmake/dependencies.cmake:1]: Normally, I would put in `cmake`. But I don't see any issue with putting it in the `scripts` directory, so I'm willing to make that change.

Comment (haruhi55) [reply cmake/dependencies.cmake:1]: Normally, I would put it in `cmake`, but I don't see any issue with putting it in the scripts directory, so I'm willing to make that change.

Comment (haruhi55) [reply cmake/dependencies.cmake:1]: done.

**Review by KuangjuX — APPROVED** [review]
LGTM!

**Pull request merged**